src/swap/changehero.ts

```diff
@@ -306,15 +306,27 @@ export function makeChangeHeroPlugin(
       ],
       networkFeeOption:
         request.fromCurrencyCode.toUpperCase() === 'BTC' ? 'high' : 'standard',
-      swapData: {
+      assetAction: {
+        assetActionType: 'swap'
+      },
+      savedAction: {
+        actionType: 'swap',
+        swapInfo,
         orderUri: orderUri + quoteInfo.id,
         orderId: quoteInfo.id,
         isEstimate: false,
+        destAsset: {
+          pluginId: request.toWallet.currencyInfo.pluginId,
+          tokenId: request.toTokenId,
+          nativeAmount: amountExpectedToNative
+        },
+        sourceAsset: {
+          pluginId: request.fromWallet.currencyInfo.pluginId,
+          tokenId: request.fromTokenId,
+          nativeAmount: amountExpectedFromNative
+        },
```

Comment:
Not a request for change, but more of an "oops, oh well":

The swap request uses the terms "from" and "to", but the swap action uses the terms "dest" and "source". It might have been sweeter to do `fromAsset: EdgeAsset` and `toAsset: EdgeAsset` in the swap action types, since it's not only shorter but also matches the existing naming.